Userland/Libraries/LibWeb/Cookie/ParsedCookie.cpp

```diff
@@ -351,3 +354,69 @@ Optional<Core::DateTime> parse_date_time(StringView date_string)
 }
 
 }
+
+bool IPC::encode(IPC::Encoder& encoder, const Web::Cookie::ParsedCookie& cookie)
+{
+    encoder << cookie.name;
+    encoder << cookie.value;
+
+    encoder << cookie.expiry_time_from_expires_attribute.has_value();
+    if (cookie.expiry_time_from_expires_attribute.has_value())
+        encoder << cookie.expiry_time_from_expires_attribute.value();
+
+    encoder << cookie.expiry_time_from_max_age_attribute.has_value();
+    if (cookie.expiry_time_from_max_age_attribute.has_value())
+        encoder << cookie.expiry_time_from_max_age_attribute.value();
+
+    encoder << cookie.domain.has_value();
+    if (cookie.domain.has_value())
+        encoder << cookie.domain.value();
+
+    encoder << cookie.path.has_value();
+    if (cookie.path.has_value())
+        encoder << cookie.path.value();
+
+    encoder << cookie.secure_attribute_present;
+    encoder << cookie.http_only_attribute_present;
+
+    return true;
+}
+
+bool IPC::decode(IPC::Decoder& decoder, Web::Cookie::ParsedCookie& cookie)
+{
+    auto decode_optional = [&](auto& optional) {
```

Comment:
Hmm, hold up, can we not `decode()` directly onto an `Optional<T>`?
`IPC::Decoder` has `template<typename T> bool decode(Optional<T>&)`

Reply:
Oh maybe, I'll check - I looked around for an `Optional` decoder but apparently not hard enough :)

Reply:
holy moly yeah `Encoder` and `Decoder` already do all this...will fix